rest_api/db/offers_query.py

```diff
@@ -18,6 +18,31 @@
 
 from api.errors import ApiBadRequest
 
+from db import blocks_query
+
+
+async def fetch_all_offer_resources(conn, query_params):
+    return await r.table('offers')\
+        .filter(query_params)\
+        .filter((blocks_query.latest_block_num() >= r.row['start_block_num'])
```

Comment:
The query parameter filter should probably come after the block number filter. My assumption is the block number filter will typically reduce the size of the list more than the query parameters.